channeldb/payments.go

```diff
@@ -957,6 +927,78 @@ func (d *DB) DeletePayments(failedOnly, failedHtlcsOnly bool) error {
 	}, func() {})
 }
 
+// deletePaymentAttempt removes an HTLC attempt from disk and returns the
+// payment in the state before it was deleted.
+func (d *DB) deletePaymentAttempt(hash lntypes.Hash, attemptID uint64) (*MPPayment, error) {
+	aid := make([]byte, 8)
+	binary.BigEndian.PutUint64(aid, attemptID)
+
+	var payment *MPPayment
+	err := kvdb.Update(d, func(tx kvdb.RwTx) error {
+		payment = nil
+		payments := tx.ReadWriteBucket(paymentsRootBucket)
+		if payments == nil {
+			return nil
+		}
+
+		bucket := payments.NestedReadWriteBucket(hash[:])
+		if bucket == nil {
+			return nil
+		}
+
+		p, err := fetchPayment(bucket)
+		payment = p
+
+		if err != nil {
```

Comment:
Usually the pattern is, we check the returned `err` first before everything else, so it's better to check `err != nil` before the above assignment.

Reply:
Done.